go.mod

```diff
@@ -4,6 +4,7 @@ require (
 	github.com/NebulousLabs/go-upnp v0.0.0-20180202185039-29b680b06c82
 	github.com/Yawning/aez v0.0.0-20211027044916-e49e68abd344
 	github.com/andybalholm/brotli v1.0.3
+	github.com/bahlo/generic-list-go v0.2.0
```

Comment:
Fair enough, though I should note that this _specific_ dependency is very small (the generic list impl is the only thing in it), it's a drop-in replacement, it's based on stdlib code, it's complete, fully tested has no open issues and hasn't had a need for any new code in over 2 years, so it's about as good a dependency as anyone could hope for :p

To bring this into the fn package, I'd just copy the subset of the code that was used here over (code is BSD licensed). Would that be ok?

Reply:
What were the changes you did on this benchmark? Because the `no_additional_results` case is significantly different from the original benchmark in the PR:

```
MissionControlStoreFlushing/no_additional_results-7      221µs ±15%       0µs ± 2%   -99.86%  (p=0.000 n=10+10)
```